crates/ln-dlc-node/src/tests/just_in_time_channel.rs

```diff
@@ -70,44 +62,108 @@ async fn just_in_time_channel() {
     // configuration value for said channel.
     let invoice_amount = 1_000;
 
+    send_interceptable_payment(
+        &payer,
+        &payee,
+        &coordinator,
+        invoice_amount,
+        JUST_IN_TIME_CHANNEL_OUTBOUND_LIQUIDITY_SAT,
+    )
+    .await
+    .unwrap();
+}
+
+/// Used to ascertain if a payment will be routed through a channel
+/// according to the
+/// `max_inbound_htlc_value_in_flight_percent_of_channel`
+/// configuration flag of the receiving end of the channel.
+fn does_inbound_htlc_fit_as_percent_of_channel(
+    receiving_node: &Node,
+    channel_id: &[u8; 32],
+    htlc_amount_sat: u64,
+) -> Result<bool> {
+    let htlc_amount_sat = Decimal::from(htlc_amount_sat);
+
+    let max_inbound_htlc_as_percent_of_channel = Decimal::from(
+        receiving_node
+            .user_config
+            .channel_handshake_config
+            .max_inbound_htlc_value_in_flight_percent_of_channel,
+    );
+
+    let channel_size_sat = receiving_node
+        .channel_manager
+        .list_channels()
+        .iter()
+        .find_map(|c| (&c.channel_id == channel_id).then_some(c.channel_value_satoshis))
+        .context("No matching channel")?;
+    let channel_size_sat = Decimal::from(channel_size_sat);
+
+    let max_inbound_htlc_sat =
+        channel_size_sat * (max_inbound_htlc_as_percent_of_channel / Decimal::ONE_HUNDRED);
+
+    Ok(htlc_amount_sat <= max_inbound_htlc_sat)
+}
+
+pub(crate) async fn send_interceptable_payment(
+    payer: &Node,
+    payee: &Node,
+    coordinator: &Node,
+    invoice_amount: u64,
+    coordinator_balance_before_payment: u64,
```

Comment:
🔧 I led you astray when I suggested this name yesterday. This is actually the `coordinator_just_in_time_channel_creation_outbound_liquidity`.

It might be clearer to pass in an `Option<u64>`, because `None` might more clearly indicate that the caller knows that no JIT channel will be created with that interceptable payment. Then the function can use `unwrap_or_default` to get either the `value` in `Some(value)` or `0`.